### DISCUSSION TITLE
Missing shadows

### DISCUSSION THREAD

**Discussion by Jureq1234** [pyvista/pyvista#1775] [Q&A]
HI,

I'm testing how do lights work in pyvistaI have added to scene mesh and positional light. I'm wondering why elements that are obscured are visible. How to make them invisible because od that they are obscured.
Regards,
Jerzy
![lightpyvista](https://user-images.githubusercontent.com/88145971/139827126-43a61dc1-c2e7-439f-a243-fafb1ad9de4d.JPG)

**Comment by MatthewFlamm** [comment]
Did you enable shadows?

https://docs.pyvista.org/api/core/lights.html?highlight=shadows#shadows

**Reply by Jureq1234** [reply]
Yes,
   plotter.add_light(light)
    plotter.enable_shadows()
    plotter.show()

**Comment by MatthewFlamm** [comment]
It might be that shadows only work on a separate mesh?  This looks like a single mesh. If so, my only other thought is to try physically based rendering.

**Reply by Jureq1234** [reply]
I have added a sphere in the scene, but it does not drop any shadows.


![spozgora](https://user-images.githubusercontent.com/88145971/140026469-49a289aa-23d0-4750-b6dc-f3c3ca4f1076.JPG)


 light2 = pv.Light(position=(567020.3000000001, 243437.33, 240.309), focal_point=None, color=None, light_type='scene light', intensity=1, positional=True,
    cone_angle=100, show_actor=True, exponent=None, shadow_attenuation=None, attenuation_values=None)
